pytential/symbolic/mappers.py

```diff
@@ -384,6 +512,46 @@ def __init__(self, default_source, default_target):
                                           default_source=default_source)
 
 
+class LocationReplacer(LocationTagger):
+    def _default_dofdesc(self, dofdesc):
+        return self.default_target
+
+    def map_int_g(self, expr):
+        return type(expr)(
+                expr.target_kernel, expr.source_kernels,
+                densities=self.operand_rec(expr.densities),
+                qbx_forced_limit=expr.qbx_forced_limit,
+                source=self.default_source, target=self.default_target,
+                kernel_arguments={
+                    name: self.operand_rec(arg_expr)
+                    for name, arg_expr in expr.kernel_arguments.items()
+                    }
+                )
+
+
+class DOFDescriptorReplacer(LocationReplacer):
+    r"""Mapper that replaces all the
+    :class:`~pytential.symbolic.primitives.DOFDescriptor`\ s in the expression
+    with the given ones.
```

Comment:
Maybe add use case to docstring?

Reply:
Added a little blurb in dfcee25. Let me know what you think.